docs/user_guide.md

```diff
@@ -0,0 +1,46 @@
+# Using S2I images
+
+When you have tried to use OpenShift builder images and found out that your application requires yet other actions for building/running then you are trying to leverage user scripts from `.s2i/bin`. To make your path smooth here we have collected some approaches and not always obvious receipts.
+
+## Invoking scripts embedded into an image
+
+Typically builder images provide its own version of [S2I scripts](builder_image.md#s2i-scripts) that are covering the most common use-cases. When these scripts don't fulfill your needs, S2I provides a way of overriding them by adding custom ones in the `.s2i/bin` directory. But doing this you are completely replacing the standard scripts. In some cases this is acceptable but what if you just need to execute a few commands before (or after) the scripts and then proceed execution flow as before? In this case it is possible to create a wrapper script that will execute custom logic and delegate further work to a standard script.
```

Comment:
s/its/their/
s/version of/version of the/
s/that are covering/that cover/
w/When these scripts/If these scripts/
s/But doing this/However, by doing this/
s/In some cases..../In some cases this is acceptable but in other scenarios you may prefer to execute a few commands before(or after) the scripts while retaining the logic of the script provided in the image."
s/to a standard script/to the default script in the image./